operational_analysis/toolkits/imputing.py

```diff
@@ -7,42 +7,44 @@
 from tqdm import tqdm
 
 
-def correlation_matrix_by_id_column(df, align_col, id_col, value_col):
+def correlation_matrix_by_id_column(align_col, id_col, value_col):
     """Create a correlation matrix between different assets in a data frame
 
     Args:
-        df(:obj:`pandas.DataFrame`): input data frame
-        align_col(:obj:`str`): name of column in <df> on which different assets are to be aligned
-        id_col(:obj:`str`): the column distinguishing the different assets
-        value_col(:obj:`str`): the column containing the data values to be used when assessing correlation
+        align_col(:obj:`array-like`): array of different assets are to be aligned
+        id_col(:obj:`array-like`): array distinguishing the different assets
+        value_col(:obj:`array-like`): array containing the data values to be used when assessing correlation
 
     Returns:
         :obj:`pandas.DataFrame`: Correlation matrix with <id_col> as index and column names
     """
-    assets = df[id_col].unique()  # List different assets in <id_col>
+    assets = np.unique(id_col)  # List different assets in <id_col>
     corr_df = pd.DataFrame(index=assets, columns=assets, dtype=float,
                            data=np.nan)  # Define correlation matrix that is set to NaN by default
+    df = pd.DataFrame(np.column_stack((id_col, align_col, value_col)), columns=["id_col", "align_col", "value_col"])
+    df.value_col = df.value_col.astype(float)
```

Comment:
For example, is this going to incur a performance penalty in the case where id_col, align_col, and value_col are coming from a separate pandas datafame?

Reply:
I would assume there is a performance hit if they are all coming from the same data `pd.DataFrame`, because we're effectively just putting the data back together again compared to three separate arrays. I assumed the intent was to make so `np.ndarray` and `pd.Series` objects could be passed into the function, or is there something I was missing?